chia/wallet/wallet_state_manager.py

```diff
@@ -923,38 +921,37 @@ async def handle_nft(
                 uncurried_nft.singleton_launcher_id.hex(),
             )
             return wallet_identifier
-        for wallet_info in await self.get_all_wallet_info_entries(wallet_type=WalletType.NFT):
-            nft_wallet_info: NFTWalletInfo = NFTWalletInfo.from_json_dict(json.loads(wallet_info.data))
-            if nft_wallet_info.did_id == old_did_id and old_derivation_record is not None:
+        for nft_wallet in self.wallets.values():
+            if not isinstance(nft_wallet, NFTWallet):
+                continue
```

Comment:
our wallet protocol has a `type()` functions, doesn't it?
Maybe it won't work as well with mypy, since it won't "cast" `nft_wallet` to the right type

Reply:
Yes the `isinstance` check makes `mypy` happy.